packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx

```diff
@@ -75,6 +77,19 @@ export class CalendarMonth extends BaseComponent<ICalendarMonthProps, {}> {
             </span>
           </div>
           <div className={ css('ms-DatePicker-currentYear js-showYearPicker', styles.currentYear) }>{ navigatedDate.getFullYear() }</div>
+          {
+            showMonthPickerAsOverlay ?
+              <div
+                className={ css('ms-DatePicker-headerToggleView js-showYearPicker', styles.headerToggleView) }
+                onClick={ () => this._onSelectSwitchCalendar(false) }
+                onKeyDown={ this._onKeyDown.bind(this, () => this._onSelectSwitchCalendar(true)) }
```

Comment:
You don't want to bind event handlers in this way, i.e., using anonymous functions inside render. Use @autobind instead, do you mind changing the others? Also in CalendarDay where applicable.